cli/main.cpp

```diff
@@ -95,6 +101,10 @@ int main(int argc, char* argv[])
     GetModuleFileNameA(nullptr, exename, sizeof(exename)/sizeof(exename[0])-1);
     argv[0] = exename;
 #endif
+#if defined(__APPLE__)
+	uint32_t size = sizeof(exename);
+	_NSGetExecutablePath(exename, &size);
```

Comment:
don't you need to assign `argv[0] = exename;` here?

Reply:
What do you mean ? argv[0] is an input value, why do you want to change it ? Do you mean `exename = argv[0]` or `strcpy(...` ?

Reply:
But your `exename` is not used for anything, is it? 

in the `_WIN32` code we assign `argv[0]` because that is then passed to `exec.check`

Reply:
Ok you're right: my PR does nothing. I'll fix it.